optimize/backend/pom.xml

```diff
@@ -100,14 +113,18 @@
       <groupId>org.eclipse.jetty.http2</groupId>
       <artifactId>jetty-http2-server</artifactId>
     </dependency>
+    <!-- unused/declared
     <dependency>
       <groupId>org.eclipse.jetty</groupId>
       <artifactId>jetty-alpn-java-client</artifactId>
     </dependency>
+    -->
+    <!-- unused/declared
     <dependency>
       <groupId>org.eclipse.jetty</groupId>
       <artifactId>jetty-alpn-java-server</artifactId>
     </dependency>
+    -->
```

Comment:
I think these were added for HTTP2 support. It would be good to confirm that their exclusion won't break things. Maybe @matthewBearCamunda can provide more context here

Reply:
These were indeed added for http2 support

However, I just did a test where I removed these dependencies, and I was still able to get http2 to work. These should be safe for removal

As part of the testing process for this ticket, I think we should also do some https & http2 tests just to confirm

Reply:
As we agreed on slack, we're keeping these as per documentation's suggestion